libmcount/mcount.c

```diff
@@ -783,15 +800,16 @@ struct mcount_thread_data *mcount_prepare(void)
 
 	compiler_barrier();
 
+	if (!mtdp)
+		mtdp = mcount_thread_data_alloc();
+
 	mcount_filter_setup(mtdp);
 	mcount_watch_setup(mtdp);
-	mtdp->rstack = xmalloc(mcount_rstack_max * sizeof(*mtd.rstack));
+	mtdp->rstack = xmalloc(mcount_rstack_max * sizeof(*mtdp->rstack));
 
 	pthread_once(&once_control, mcount_init_file);
 	prepare_shmem_buffer(mtdp);
 
-	pthread_setspecific(mtd_key, mtdp);
```

Comment:
With this change, `mtd_dtor()` won't get called when a thread exits.  We rely on this to synchronize thread status with uftrace process and release the resources.  So even if we don't use the TSD APIs anymore, we should keep it for this reason.